### PR TITLE
Refactor ConfigurationContainer

### PR DESCRIPTION
**Pull request by skarahoda**
- Split the main logic to four pages: Home, Login, ConfigurationCreate, and ConfigurationUpdate
- Home page only decides the which page should be the next step
- Common hooks and components are under `src/hooks` and `src/components` folders
- Split the hook `useConfigurationSave` into two hooks: `useConfigurationCreate` and `useConfigurationUpdate`

### CODE REVIEW

File: src/client/containers/ConfigurationCreateContainer/hooks/useConfigurationCreate.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
+import { useEffect } from "react";
+import { useMutation } from "react-query";
+import * as microsoftTeams from "@microsoft/teams-js";
+
+import { requester } from "../../../lib";
+import { Resource, ResourceType, WebhookEventType } from "../../../constants";
+import { errorToText } from "./errorToText";
```
Comment (dirtybit) [inline src/client/containers/ConfigurationCreateContainer/hooks/useConfigurationCreate.ts:7]: `errorToText` modules (embedded below) need to be reworked:
 - Make sure that they contain only the relevant error messages to transform
 - Correct wording: For example, `Only project members can update integrations settings.` should be rephrased for `ConfigurationCreateContainer`, i.e., `Only project members can setup integrations.` etc.


https://github.com/zeplin/microsoft-teams-app/blob/2b57ed80561a048ab1bb9f2c32d77f586deb7a03/src/client/containers/ConfigurationCreateContainer/hooks/errorToText.ts#L12-L36

https://github.com/zeplin/microsoft-teams-app/blob/2b57ed80561a048ab1bb9f2c32d77f586deb7a03/src/client/containers/ConfigurationUpdateContainer/hooks/errorToText.ts#L12-L36

File: src/client/containers/HomeContainer/HomeContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    const getUrl = (): string => {
+        const searchParams = new URLSearchParams(query as Record<string, string>).toString();
+        if (!storage.getAccessToken()) {
+            return `/login?${searchParams}`;
+        }
+        if (id) {
+            return `/configuration/update?${searchParams}`;
+        }
+        return `/configuration/create?${searchParams}`;
+    };
```
Comment (dirtybit) [inline src/client/containers/HomeContainer/HomeContainer.tsx:19]: I think it would be good to move all URL construction logic to a central place and use that module to build URLs via parameters instead of hard coding urls and search parameters. For example `urls.ts` module:
```js
{
   configurationUpdate({ channel, id }) { .. },
   configurationCreate({ channel, id }) { .. }
}
```

PS: This comment applies to all places where we build URLs (login container, configuration containers etc.).

File: src/client/containers/ConfigurationCreateContainer/hooks/useConfigurationCreate.ts (hunk 1 above)
Comment (skarahoda) [reply src/client/containers/ConfigurationCreateContainer/hooks/useConfigurationCreate.ts]: 👌

File: src/client/containers/HomeContainer/HomeContainer.tsx (hunk 2 above)
Comment (skarahoda) [reply src/client/containers/HomeContainer/HomeContainer.tsx:18]: 👌

File: src/client/containers/ConfigurationCreateContainer/hooks/useConfigurationCreate.ts [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            return "Only organization editor (or higher) can setup integration settings.";
+        case "Only owner of the project can access webhooks":
+            return "Only owner of the project can setup integration settings.";
+        case "Only owner of the styleguide can access webhooks":
+            return "Only owner of the styleguide can setup integration settings.";
+        case "Cannot create a webhook with active status because provided URL is unhealthy":
+        case "Owner of the webhook not found. Please contact to support@zeplin.io":
+        default:
+            return "We're experiencing an issue here. Please try it later or let us know: support@zeplin.io.";
+    }
```
Comment (dirtybit) [inline src/client/containers/ConfigurationCreateContainer/hooks/useConfigurationCreate.ts:36]:
```suggestion
    switch (error instanceof ClientError ? error.message : "") {
        case "User is not a member of the project":
            return "Only project members can setup integrations.";
        case "User is not a member of the styleguide":
            return "Only styleguide members can setup integrations.";
        case "Project not found":
        case "Project is archived":
            return "Project is not available anymore. You can select different Project/Styleguide to setup configuration.";
        case "Styleguide not found":
        case "Styleguide is archived":
            return "Styleguide is not available anymore. You can select different Project/Styleguide to setup configuration.";
        case "Only organization editor (or higher) can access project webhooks":
        case "Only organization editor (or higher) can access styleguide webhooks":
            return "Only organization editor (or higher) can setup integrations.";
        case "Only owner of the project can access webhooks":
            return "Only owner of the project can setup integrations.";
        case "Only owner of the styleguide can access webhooks":
            return "Only owner of the styleguide can setup integrations.";
        case "Cannot create a webhook with active status because provided URL is unhealthy":
        case "Owner of the webhook not found. Please contact to support@zeplin.io":
        default:
            return "We're experiencing an issue here. Please try it later or let us know: support@zeplin.io.";
    }
```

File: src/client/containers/ConfigurationCreateContainer/hooks/useConfigurationCreate.ts [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+};
+
+function setSettings(configurationId: string, resourceName: string, resourceType: ResourceType): void {
+    const contentURL = new URL(window.location.href);
+    contentURL.pathname = "/";
+    contentURL.searchParams.set("id", configurationId);
+    contentURL.searchParams.set("theme", "{theme}");
+    contentURL.searchParams.set("channel", "{channelName}");
+    contentURL.searchParams.set("resourceName", resourceName);
+    contentURL.searchParams.set("resourceType", resourceType);
```
Comment (dirtybit) [inline src/client/containers/ConfigurationCreateContainer/hooks/useConfigurationCreate.ts:46]: This can be moved to the url module as well.

File: src/client/lib/url.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+        return `/login?${window.location.search}`;
+    }
+
+    get configurationCreate(): string {
+        return `/configuration/create?${window.location.search}`;
+    }
+
+    get configurationUpdate(): string {
+        return `/configuration/update?${window.location.search}`;
+    }
```
Comment (dirtybit) [inline src/client/lib/url.ts:12]: I think we should prefer being explicit for query parameters while building these URLs.  

As a person reading this could, I should be able to understand what parameters can be passed to these URLs and if they are required or not. Otherwise it gets obfuscated and hard to follow, hence gets more error prone.

**Review by dirtybit — APPROVED** [review]
